### PR TITLE
Hacktoberfest: Add a solution for Project Euler 49

### PR DESCRIPTION
**Pull request by Iqrar99**
Seperated the PR based on #2548
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: project_euler/problem_49/sol1.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                    if found:
+                        break
+                if found:
+                    break
+            if found:
+                break
+
+    answer = set()
+    for seq in passed:
+        answer.add("".join(list(map(str, seq))))
```
Comment (dhruvmanila) [inline project_euler/problem_49/sol1.py:129]:
```suggestion
        answer.add("".join([str(i) for i in seq]))
```
As mentioned in CONTRIBUTING.md

File: project_euler/problem_49/sol1.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+    return max(map(int, [x for x in answer]))
```
Comment (dhruvmanila) [inline project_euler/problem_49/sol1.py:131]:
```suggestion
    return max([int(x) for x in answer]))
```
`map` in here was completely unnecessary.

**Review by dhruvmanila — CHANGES_REQUESTED** [review]

File: project_euler/problem_49/sol1.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+(ii) each of the 4-digit numbers are permutations of one another.
+
+There are no arithmetic sequences made up of three 1-, 2-, or 3-digit primes,
+exhibiting this property, but there is one other 4-digit increasing sequence.
+
+What 12-digit number do you form by concatenating the three terms in this sequence?
+"""
+
+"""
+Solution:
```
[the remark this answers is not in the document]
Comment (Iqrar99) [reply project_euler/problem_49/sol1.py:21]: Ah ok

**Comment by Iqrar99** [comment]
Thanks for the review @dhruvmanila. I'll make the changes soon.

File: project_euler/problem_49/sol1.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+def is_prime(number: int) -> bool:
+    """
+    function to check whether the number is prime or not.
+    >>> is_prime(2)
+    True
+    >>> is_prime(6)
+    False
+    >>> is_prime(1)
+    False
```
[the remark this answers is not in the document]
Comment (Iqrar99) [reply project_euler/problem_49/sol1.py:41]: sure

File: project_euler/problem_49/sol1.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+def solution():
+    """
+    Return the solution of the problem.
+    >>> solution()
+    296962999629
+    """
+    prime_list = [n for n in range(1001, 10000, 2) if is_prime(n)]
+    candidates = []
+
+    for x in prime_list:
```
[the remark this answers is not in the document]
Comment (Iqrar99) [reply project_euler/problem_49/sol1.py:85]: ok

File: project_euler/problem_49/sol1.py [hunk 6]
```diff
@@ -0,0 +1,140 @@
+from itertools import permutations
+from math import floor, sqrt
+
```
Comment (dhruvmanila) [inline project_euler/problem_49/sol1.py:3]: Oh, did not see that. Please put your imports after the module docstring.

**Review by dhruvmanila — CHANGES_REQUESTED** [review]

File: project_euler/problem_49/sol1.py (hunk 6 above)
Comment (Iqrar99) [reply project_euler/problem_49/sol1.py]: alright

File: project_euler/problem_49/sol1.py [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    """
+    prime_list = [n for n in range(1001, 10000, 2) if is_prime(n)]
+    candidates = []
+
+    for number in prime_list:
+        perm = list(permutations(list(str(number))))
+        tmp_numbers = []
+
+        for prime_member in perm:
+            prime = int("".join(list(prime_member)))
```
Comment (dhruvmanila) [inline project_euler/problem_49/sol1.py:95]:
```suggestion
        tmp_numbers = []

        for prime_member in permutations(list(str(number))):
            prime = int("".join(prime_member))
```

**Review by dhruvmanila — CHANGES_REQUESTED** [review]
Looks good to me! Thank you for your contribution. 🎉 😃

**Comment by Iqrar99** [comment]
> Looks good to me! Thank you for your contribution. 🎉 😃

You are welcome! Maybe you want to review my Project Euler 50 PR? @dhruvmanila here #2703

**Comment by dhruvmanila** [comment]
Sure thing, but please keep this in mind that we receive a lot of PRs and if possible only submit one PR at a time and once that is approved, submit another one.

***The ocean is made up of tiny droplets of water***

**Comment by Iqrar99** [comment]
I see. Take your time 😄

**Comment by dhruvmanila** [comment]
Are you participating in Hacktoberfest? If so, please include ***for Hacktoberfest*** in the description of your PR to signal us.

**Review by dhruvmanila — APPROVED** [review]
#2695

**Comment by Iqrar99** [comment]
Thank you so much 😄  I'm looking forward to contributing more!

**Pull request merged**